pkg/store/proxy.go

```diff
@@ -515,6 +549,22 @@ func (s *ProxyStore) LabelValues(ctx context.Context, r *storepb.LabelValuesRequ
 		span           opentracing.Span
 	)
 
+	// We may arrive here either via the promql engine
+	// or as a result of a grpc call in layered queries
+	var tenant string
+	var err error
+	if gctx.Value(tenancy.TenantKey) != nil {
+		tenant = gctx.Value(tenancy.TenantKey).(string)
+	} else {
+		tenant, err = tenancy.GetTenantFromGRPCMetadata(ctx)
```

Comment:
Not sure I understand this. gctx seems to be derived from the same ctx?

Reply:
Right, I guess it doesn't really matter if we use `ctx` or `gctx` here, both will have the needed info. Probably ought to use `gctx` ?

We still need to check both the context directly for the tenant information or via the grpc metadata, depending on how we end up here.

Reply:
Updated to use `gctx` everywhere.